src/treeinterp/BTFactory.cpp

```diff
@@ -19,6 +28,21 @@ void BTFactory::makeTrees() {
 
     std::cout << "Re-Make Trees From Json" << std::endl;
 
+    // We need this instance of the TreeProtoType Class to use its methods. An alternative would be to make the methods static,
+    // but this essentially makes them global which is dangerous and unnecessary in my opinion. Additionally, some settings
+    // are initialized in the default constructor of TreeProtoType
+    bt::TreeProtoType protoTypeTree;
+    std::shared_ptr<bt::BehaviorTree> testingTree;
+    testingTree = protoTypeTree.createOffensiveStrategy();
```

Comment:
Static methods != global methods if they are properly handled using includes. Global objects are very bad because they state you cannot reason about. " Global" static methods are actually quite fine since they do not have any implicit state.

Also, I think the treeProtoType does not need to be a class here but it can just be a method. I feel the name is very misleading, and it's not really clear what happens when you construct a 'TreeProtoType' .

Reply:
I think it can be improved by renaming the treeprototype class to, in this case, OffensiveStrategy, and then having a create method for it, I think this would remedy the confusing name. I see your argument for making the methods static, I think this does remove unnecessary code. So having something like OffensiveStrategy::create() seems ok to me, but I am curious what you think about this.

Reply:
I've changed my mind about this, as bypassing the constructor means you can't give the OffensiveStrategy the tree it makes since the object itself is never created, and some stuff also needs to be done in the constructor